### PR TITLE
[IMP] account, sale: Hide tax column for US & BR based companies

### PR DESCRIPTION
**Pull request by MuhamedAbdallaMuhamed**
[IMP] account, sale: Hide tax column for US & BR based companies

in some scenarios users can see several taxes listed in the "Taxes" column in the PDF report (for SO, quotes, invoices, etc).
Having the column oversaturated with all the taxes can represent a problem for our users.

it's a pain currently and we predict that we are going to receive this kind of request also for other countries
and in case of mutlti-company we enforce the company be a USA/Brazil country base

Solution:
1- Add new stored column to control showing the tax column through allowing any module to overwrite it.
2- Use the new column 'show_tax_column' to hide the tax column in portal preview and PDF report for US/BR that has avatax active

Task-3864811

enterprise-pr#https://github.com/odoo/enterprise/pull/65409

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/views/sale_portal_templates.xml [hunk 1]
```diff
@@ -573,7 +574,7 @@
                                                 <t t-out="((line.discount % 1) and '%s' or '%d') % line.discount"/>%
                                             </strong>
                                         </td>
-                                        <td t-attf-class="text-end {{ 'd-none d-md-table-cell' if report_type == 'html' else '' }}" id="taxes">
+                                        <td t-attf-class="text-end {{ 'd-none' if not show_invoice_tax else '' }} {{ 'd-none d-md-table-cell' if report_type == 'html' and show_invoice_tax else '' }}" id="taxes">
```
Comment (Feyensv) [inline addons/sale/views/sale_portal_templates.xml:577]: Why not a simple `t-if` on the `td` ?
Is there a reason to add invisible content to the report ?

Comment (muab-odoo) [reply addons/sale/views/sale_portal_templates.xml]: Fixed!

File: addons/sale/report/ir_actions_report_templates.xml [hunk 2]
```diff
@@ -80,7 +81,9 @@
                         <th name="th_discount" t-if="display_discount" class="text-end">
                             <span>Disc.%</span>
                         </th>
-                        <th name="th_taxes" class="text-end">Taxes</th>
+                        <div t-if="show_invoice_tax">
+                            <th name="th_taxes" class="text-end">Taxes</th>
+                        </div>
```
Comment (Feyensv) [inline addons/sale/report/ir_actions_report_templates.xml:86]:
```suggestion
                        <th t-if="show_invoice_tax" name="th_taxes" class="text-end">Taxes</th>
```

File: addons/sale/report/ir_actions_report_templates.xml [hunk 3]
```diff
@@ -110,10 +113,12 @@
...
-                                <t t-set="taxes" t-value="', '.join([(tax.invoice_label or tax.name) for tax in line.tax_id])"/>
-                                <td name="td_taxes" t-attf-class="text-end {{ 'text-nowrap' if len(taxes) &lt; 10 else '' }}">
-                                    <span t-out="taxes">Tax 15%</span>
-                                </td>
+                                <div t-if="show_invoice_tax">
+                                    <t t-set="taxes" t-value="', '.join([(tax.invoice_label or tax.name) for tax in line.tax_id])"/>
+                                    <td name="td_taxes" t-attf-class="text-end {{ 'text-nowrap' if len(taxes) &lt; 10 else '' }}">
+                                        <span t-out="taxes">Tax 15%</span>
+                                    </td>
+                                </div>
```
Comment (Feyensv) [inline addons/sale/report/ir_actions_report_templates.xml:121]:
```suggestion
                                <t t-if="show_invoice_tax">
                                    <t t-set="taxes" t-value="', '.join([(tax.invoice_label or tax.name) for tax in line.tax_id])"/>
                                    <td name="td_taxes" t-attf-class="text-end {{ 'text-nowrap' if len(taxes) &lt; 10 else '' }}">
                                        <span t-out="taxes">Tax 15%</span>
                                    </td>
                                </t>
```

File: addons/sale/report/ir_actions_report_templates.xml [hunk 4]
```diff
@@ -67,6 +67,7 @@
 
             <!-- Is there a discount on at least one line? -->
             <t t-set="lines_to_report" t-value="doc._get_order_lines_to_report()"/>
+            <t t-set="show_invoice_tax" t-value="doc.company_id.show_invoice_tax"/>
```
Comment (Feyensv) [inline addons/sale/report/ir_actions_report_templates.xml:70]:
```suggestion
            <t t-set="show_taxes" t-value="doc.company_id.show_invoice_tax"/>
```

**Review by Feyensv — APPROVED** [review]

**Review by qdp-odoo — APPROVED** [review]

**Pull request closed without merging**